packages/tools/devtools/devtools-core/src/data-visualization/SharedTreeVisualizer.ts

```diff
@@ -302,3 +368,53 @@ export async function visualizeSharedTreeNodeBySchema(
 		}
 	}
 }
+
+/**
+ * Creates a visual representation of a SharedTree based on its schema.
+ * @param tree - The {@link VerboseTree} to visualize
+ * @param treeDefinitions - The schema that defines the structure and types of the tree
+ * @param allowedTypes - String representing the allowed types for this node
+ * @param isRequired - Whether this field is required in its parent object schema.
+ * Only meaningful for direct children of object nodes.
+ * Undefined for array/map elements since they are always required within their parent.
+ * @param visualizeChildData - Callback function to visualize child node data
+ * @returns A visual representation of the tree that includes schema information and node values
+ *
+ * @remarks
+ * This function handles both leaf nodes (primitive values, handles) and internal nodes (objects, maps, arrays).
+ * For leaf nodes, it creates a visual representation with the node's schema and value.
+ * For internal nodes, it recursively processes the node's fields using {@link visualizeInternalNodeBySchema}.
+ */
+export async function visualizeSharedTreeBySchema(
+	tree: VerboseTree,
+	treeDefinitions: ReadonlyMap<string, SimpleNodeSchema>,
+	allowedTypes: string,
+	isRequired: string | undefined,
+	visualizeChildData: VisualizeChildData,
+): Promise<VisualSharedTreeNode> {
+	const schemaFactory = new SchemaFactory(undefined);
+
+	return Tree.is(tree, [
```

Comment:
Just double-checking: is there an existing "isLeaf" (or something) helper we could use for this check?

Reply:
Will double check on this. Thanks for the reminder.

Reply:
There is [treeValue()](https://github.com/microsoft/FluidFramework/blob/0052d4cc85ef19c99215644d91e9a96869b5f6e4/packages/dds/tree/src/feature-libraries/valueUtilities.ts#L66) utiliy function that is consumed internally in the ST package, but I am not sure if I can expose this to outside package.

Reply:
Note that function doesn't handle Handle nodes.

@CraigMacomber maybe you can advise on the best publicly available pattern for checking if a node is a leaf using the VerboseTree APIs?

Reply:
Since leaves in VerboseTree are the same as leaves in regular trees. you can do something like:
```
const f = new SchemaFactory(undefined);
const leaf = [f.number, f.string, f.null, f.handle, f.boolean];
Tree.is(node, leaf);
``

Reply:
Not that I expect us to add a new leaf type anytime soon, but it seems like it would be nice to have this sort of check encapsulated in a single place so we don't have to update all of these hardcoded lists if/when we do add a new leaf type. Not for this PR. but would it be worthwhile to add a centralized `isLeaf` function for this?